File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start=0
        hashset=set()
        res=0
        for end in range(len(s)):
            while s[end] in hashset:
                hashset.remove(s[start])
                start+=1
            hashset.add(s[end])
            res=max(res,end-start+1)
            print(hashset)
        return res

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var uniqueCharSet=new HashSet<char>();
        int maxLen=0;
        int st=0;
        int end=0;

        while(end<s.Length){
            while(uniqueCharSet.Contains(s[end])){
                uniqueCharSet.Remove(s[st]);
                st++;
            }
            uniqueCharSet.Add(s[end]);
            maxLen=Math.Max(maxLen,end-st+1);
            end++;
        }
        return maxLen;

    }
}